### ISSUE TITLE
Add e2e test coverage for relay after ibc genesis restart upgrade

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#3476]
## Summary

After #3241 we can add additional test coverage to complete a relay after the genesis restart upgrade. [Hermes](https://github.com/informalsystems/hermes/issues/1152#issuecomment-1499192815) now supports this functionality. I recommend doing in a separate pr since it might require non-trivial relayer specific logic to complete this issue

see [comment](https://github.com/cosmos/ibc-go/issues/3241#issuecomment-1460110656)

To complete this issue we should use relayer specific logic to allow the client to be updated after a genesis restart. After the genesis restart completes we want to test that the client can be updated and that a packet can be relayed. It should verify the transfer and ics27 logic succeeded 

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
